jerry-core/api/jerry.c

```diff
@@ -5110,6 +5110,122 @@ jerry_get_container_type (const jerry_value_t value) /**< the container object *
+    {
+      uint16_t type = ((ecma_extended_object_t *) obj_p)->u.class_prop.class_id;
+      ecma_extended_object_t *container_object_p = ecma_op_container_get_object (container, type);
+
+      if (container_object_p == NULL)
+      {
+        return ecma_create_error_reference_from_context ();
+      }
+
+      jerry_value_t result;
+
+      switch (operation)
+      {
+#if ENABLED (JERRY_BUILTIN_SET) || ENABLED (JERRY_BUILTIN_WEAKSET)
+        case JERRY_CONTAINER_OP_ADD:
+        {
+          result = jerry_return (ecma_op_container_set (container_object_p, arguments[0], arguments[0], type));
+          break;
+        }
+#endif /* ENABLED (JERRY_BUILTIN_SET) || ENABLED (JERRY_BUILTIN_WEAKSET) */
+#if ENABLED (JERRY_BUILTIN_MAP) || ENABLED (JERRY_BUILTIN_WEAKMAP)
+        case JERRY_CONTAINER_OP_GET:
+        {
+          result = jerry_return (ecma_op_container_get (container_object_p, arguments[0], type));
+          break;
+        }
+        case JERRY_CONTAINER_OP_SET:
+        {
+          result = ecma_op_container_set (container_object_p, arguments[0], arguments[1], type);
+
+          if (!jerry_value_is_error (result))
+          {
+            jerry_release_value (result);
+            result = jerry_create_boolean (true);
+          }
+          break;
+        }
+#endif /* ENABLED (JERRY_BUILTIN_MAP) || ENABLED (JERRY_BUILTIN_WEAKMAP) */
+#if ENABLED (JERRY_BUILTIN_SET) || ENABLED (JERRY_BUILTIN_WEAKSET) \
+        || ENABLED (JERRY_BUILTIN_MAP) || ENABLED (JERRY_BUILTIN_WEAKMAP)
+        case JERRY_CONTAINER_OP_HAS:
+        {
+          result = jerry_return (ecma_op_container_has (container_object_p, arguments[0], type));
+          break;
+        }
+        case JERRY_CONTAINER_OP_DELETE:
+        {
+          if (type == LIT_MAGIC_STRING_WEAKMAP_UL || type  == LIT_MAGIC_STRING_WEAKSET_UL)
+          {
+            result = jerry_return (ecma_op_container_delete_weak (container_object_p, arguments[0], type));
+            break;
+          }
+          result = jerry_return (ecma_op_container_delete (container_object_p, arguments[0], type));
+          break;
+        }
+        case JERRY_CONTAINER_OP_SIZE:
+        {
+          result = jerry_return (ecma_op_container_size (container_object_p));
+          break;
+        }
+        case JERRY_CONTAINER_OP_CLEAR:
+        {
+          result = jerry_return (ecma_op_container_clear (container_object_p));
+          break;
+        }
+#endif
+        default:
+        {
+          result = jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG ("Unsupported container operation")));
+        }
+      }
+      return result;
+    }
+#endif /* ENABLED (JERRY_BUILTIN_CONTAINER) */
+  }
+  else
+  {
+    JERRY_UNUSED (operation);
+    JERRY_UNUSED (container);
+    JERRY_UNUSED (arguments);
+    JERRY_UNUSED (arguments_number);
+    return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG ("Container is not a object.")));;
```

Comment:
If this else is here, when will the lines below it run? This `else` is only required if the containers are enabled, in every other case the normal "disabled" error can be thrown.